datastore/google/cloud/datastore/client.py

```diff
@@ -648,3 +648,33 @@ def do_something(entity):
         if "namespace" not in kwargs:
             kwargs["namespace"] = self.namespace
         return Query(self, **kwargs)
+
+    def reserve_ids(self, complete_key, num_ids):
+        """Prevents the supplied keys' IDs from being auto-allocated by Cloud
+            Datastore.
+
+        :type complete_key: :class:`google.cloud.datastore.key.Key`
+        :param complete_key: Partial key to use as base for reserved IDs.
+
+        :type num_ids: int
+        :param num_ids: The number of IDs to reserve.
+
+        :rtype: list of :class:`google.cloud.datastore.key.Key`
+        :returns: []
```

Comment:
`:rtype:` and `:returns:` need to match the actual returned value below.  Because the `ReserveIdsResponse` message is empty, we should just be returning `None`.